docs/src/tutorials/getting_started/design_patterns_for_larger_models.jl

```diff
@@ -0,0 +1,610 @@
+# Copyright (c) 2021 Oscar Dowson and contributors                               #src
+#                                                                                #src
+# Permission is hereby granted, free of charge, to any person obtaining a copy   #src
+# of this software and associated documentation files (the "Software"), to deal  #src
+# in the Software without restriction, including without limitation the rights   #src
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell      #src
+# copies of the Software, and to permit persons to whom the Software is          #src
+# furnished to do so, subject to the following conditions:                       #src
+#                                                                                #src
+# The above copyright notice and this permission notice shall be included in all #src
+# copies or substantial portions of the Software.                                #src
+#                                                                                #src
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR     #src
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,       #src
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE    #src
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER         #src
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,  #src
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE  #src
+# SOFTWARE.                                                                      #src
+
+# # Design patterns for larger models
+
+# JuMP makes it easy to build and solve optimization models. However, once you
+# start to construct larger models, and especially ones that interact with
+# external data sources or have customizable sets of variables and constraints
+# based on client choices, you may find that your scripts become unwieldy. This
+# tutorial demonstrates a variety of ways in which you can structure larger JuMP
+# models to improve their readability and maintainability.
+
+# !!! tip
+#     This tutorial is more advanced than the other "Getting started" tutorials.
+#     It's in the "Getting started" section to give you an early preview of how
+#     JuMP makes it easy to structure larger models. However, if you are new to
+#     JuMP you may want to briefly skim the tutorial, and come back to it once
+#     you have written a few JuMP models.
+
+# ## Overview
+
+# This tutorial uses explanation-by-example. We're going to start with a simple
+# [knapsack model](https://en.wikipedia.org/wiki/Knapsack_problem), and then
+# expand it to add various features and structure.
+
```

Comment:
Add a note on how to of how to find the "data/" folder? 
https://github.com/jump-dev/JuMP.jl/blob/master/docs/src/tutorials/getting_started/data